web/src/components/tasks/business-formation/contacts/Addresses.test.tsx

```diff
@@ -144,8 +151,15 @@ describe("Formation - Addresses", () => {
           ];
           const page = await getPageHelper({ legalStructureId }, { incorporators });
           const nameTd = screen.getByText(incorporators[1].name, { exact: false });
+
           // eslint-disable-next-line testing-library/no-node-access
-          fireEvent.click(nameTd.parentElement?.querySelector('button[aria-label="delete"]') as Element);
+          const tr = nameTd.closest("tr");
+
+          if (tr !== null) {
+            const removeButton = within(tr).getByText(Config.formation.fields.signers.signDeleteLabel);
```

Comment:
```suggestion
            const removeButton = within(tr).getByRole("button", {name: Config.formation.fields.signers.signDeleteLabel"});
```
[dust]: You could opt for `getByRole` here (and other places) to enforce some accessibility.

Reply:
Nice, good callout